### PR TITLE
feat: support middleware CDP changes

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR adds a batch ID to all OTP requests and a new confirmation dialog on the terms of storage checkbox. 

These changes are done to match the functionality of https://github.com/ibi-group/otp-middleware/pull/152

### CODE REVIEW

File: i18n/en-US.yml [hunk 1]
```diff
@@ -295,6 +295,10 @@ components:
     termsOfStorageStatement: >
       Optional: I consent to the Trip Planner storing my historical planned trips in order to
       improve transit services in my area. <termsOfStorageLink>More info...</termsOfStorageLink>
+    confirmDeletionPrompt: >
+      Removing your consent to storage of historical trips will cause all data tied to you to be deleted. 
+      All your other saved locations, trips, and notifications will also be deleted permanently.
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply i18n/en-US.yml:300]: I have checked with Rob who confirmed that this is already implemented.

File: lib/actions/api.js [hunk 2]
```diff
@@ -185,7 +185,7 @@ export function routingQuery (searchId = null, updateSearchInReducer = false) {
         user.loggedInUser.storeTripHistory
 
       const nonRealtimeFetch = fetch(
-        constructRoutingQuery(state, true),
+        constructRoutingQuery(state, true, { batchId: searchId }),
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/actions/api.js]: I worry how many bugs has caused in the many years this has been missing! Either way, it's fixed in 72af4a62991108e4a1f3bd2ea5ad09bcbf0807cd

**Comment by miles-grant-ibigroup** [comment]
Thanks Binh, all changes are addressed.

**Comment by miles-grant-ibigroup** [comment]
Deletion prompt updated to match what really happens in 0c64c349e1b49f93a5919b4461d8f7604717a270

**Comment by miles-grant-ibigroup** [comment]
Also happy to confirm that these changes fix the flex request spray we've been seeing!

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**